### ISSUE TITLE
Database Error after making an allocation 

### ISSUE THREAD

**Issue by Ecoxl** [Ecoxl/celero#88]
![Celero Error Database 07052021](https://user-images.githubusercontent.com/9587988/117459795-133be600-af4c-11eb-9ff6-f34eed3bd088.JPG)
Error appeared in the Celero project "space clinker cement", company  "space clikner", process "co-processing", flow "petcoke" after making the allocation of the flow "petcoke" to the procees" co-processing". 
I have  added Basil Gisi as sonslatatnt to the project and company. 
![Celero Allocation error 07052021](https://user-images.githubusercontent.com/9587988/117462114-978f6880-af4e-11eb-8fc0-3e84f153d06e.JPG)
 As a consequence on the page  "KPI view and edit Table" empft fields for options were created.

**Comment by Ocimum-basilicum** [comment]
So this is a really strange bug which is caused by wrong used foreign keys and auto increment ids in the database.  

The are three database tables involved: 

1. t_prcss which stores all process names
2. t_cmpny_prcss which stores the processes which are used by a company
3. t_cp_allocation which stores the allocation data 

![Screenshot 2021-05-18 at 21 08 39](https://user-images.githubusercontent.com/17066673/118711149-8a605c80-b81f-11eb-9a99-e8fc7d771e1c.png)

The "t_cmpny_prcss" table is the link between the table "t_prcss" where all processes names are stored and the allocation table stores all allocations data inclusive the selected process (blue arrow). 
Now the "t_cmpny_prcss" table has a column id (id key which auto increments, every new row gets +1 in the id column) and in this table there is also the process id (foreign key constraint to table t_prcss).  

Now when creating a new allocation the software gets the "process id" to write into "t_cp_allocation" from "t_cmpny_prcss". But instead of taking the actual process id stored in column "t_cmpny_prcss.**prcss_id**" it gets the "t_cmpny_prcss.**id**" (see red arrow in the image). Which is the wrong id and not the supposed value, which is critical because its a foreign key constraint! But until now this wasn't a problem, the software was coded to work like this and provided correct outputs and the wrong ids where accepted.  But now, by creating more and more processes in each company, we reached a "t_cmpny_prcss.id" value which is larger than the "t_prcss.id". Because we want to write this "t_cmpny_prcss.id" into the "t_cp_allocation.prcss_id" which is a foreign key and only accepts values which are in "t_prcss.id" we ran into this problem! (see error message: "Key (prcss_id)=(430) is not present in table "t_prcss".) 

![Screenshot 2021-05-18 at 21 30 58](https://user-images.githubusercontent.com/17066673/118711856-589bc580-b820-11eb-97b6-c253cb772d24.png)
This screen shot tries to explain the error.

Now what I did to fix this: I created several processes named "x" (x was chosen that they are at the end of the process list) to increase the t_prcss.id to be larger as t_cmpny_prcss.id again. As long t_cmpny_prcss.id < t_prcss.id it should work as always. I know this is a silly fix but I don't have time to change and test the whole allocation pages and everything that references to it.

TL:DR if t_cmpny_prcss.id > t_prcss.id the allocation creation will not work because the t_cmpny_prcss.id is written into t_cp_allocation.prcss_id and this is a foreign key to t_prcss.id and will not accept the value.
